samza-core/src/main/java/org/apache/samza/coordinator/JobCoordinatorFactory.java

```diff
@@ -26,11 +26,12 @@
 @InterfaceStability.Evolving
 public interface JobCoordinatorFactory {
   /**
+   * Return a new instance of {@link JobCoordinator}
    * @param processorId {@link org.apache.samza.processor.StreamProcessor} id
    * @param config Configs relevant for the JobCoordinator TODO: Separate JC related configs into a "JobCoordinatorConfig"
    * @param containerController Controller interface for starting and stopping container. In future, it may simply
    *                            pause the container and add/remove tasks
-   * @return An instance of IJobCoordinator
+   * @return {@link JobCoordinator} instance
    */
   JobCoordinator getJobCoordinator(String processorId, Config config, SamzaContainerController containerController);
```

Comment:
The processorId parameter shouldn't be here, or in the StreamProcessor constructor. Similarly, LocalApplicationRunner shouldn't be using ProcessorIdGenerator itself (see explanation in previous PR). Let me know if that doesn't make sense.